test/Spec.hs

```diff
@@ -44,6 +45,8 @@ tests :: Bool -> TestTree
 tests docker0 = testGroup "Tests"
   [ testCase "Doctest Z module" (doctest ["src/Test/StateMachine/Z.hs"])
   , ShrinkingProps.tests
+  , testProperty "Towers of Hanoi"
+      (expectFailure (prop_hanoi 3))
```

Comment:
Does it always find the solution in 100 tries? If not please use `withMaxSuccess` like below to increase the number of tries.

Reply:
With 3 discs yes, so far it have not failed to find one for me, seems reliable. For 4 discs it would probably need around 2000 tries.